### PR TITLE
For #278: Added security to employee views

### PR DESCRIPTION
**Pull request by paulodamaso**
For #278: Added security to employee views
- Made employee views extend `SecuredView`;
- Added tests asserting that unauthorized users can't access employee views;
- Added skip annotation to tests because they start to fail now (no authenticated users)
- Left puzzle to continue iomplementation

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#436](https://github.com/timelesslounge/timelessis/issues/436) is now in scope, role is `REV`

**Comment by paulodamaso** [comment]
@0crat wait for #436 review and merge

**Comment by 0crat** [comment]
> @0crat wait for #436 review and merge ([here](https://github.com/timelesslounge/timelessis/issues/436#issuecomment-468014829))

@paulodamaso It's a code review job [#436](https://github.com/timelesslounge/timelessis/issues/436), can't put it on hold

**Comment by 0crat** [comment]
This pull request [#436](https://github.com/timelesslounge/timelessis/issues/436) is assigned to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/5c7b5be0-5d39-48ca-bb13-6d8dad6711d8); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: tests/integration/it_employees_test.py [hunk 1]
```diff
@@ -76,6 +83,16 @@ def test_list(client, db_session):
     assert str.encode(employee.username) in response.data
 
 
+def test_cannot_access_list(client, db_session):
+    """ Show 403 - Forbidden when user cannot access employee list """
+    company = factories.CompanyFactory()
+    manager_role = factories.RoleFactory()
+    employee = factories.EmployeeFactory(company=company, role=manager_role)
```
Comment (amihaiemil) [inline tests/integration/it_employees_test.py]: @paulodamaso Please inline the creation of the company and manager role :)

File: tests/integration/it_employees_test.py [hunk 2]
```diff
@@ -7,6 +7,12 @@
 from tests import factories
 from timeless.employees.models import Employee
 
+"""
+@ #278:30min Tests are not using secured views. We should authenticate users to
```
Comment (amihaiemil) [inline tests/integration/it_employees_test.py]: @paulodamaso puzzle format is wrong.

File: timeless/access_control/authorization.py [hunk 3]
```diff
@@ -17,12 +17,14 @@ def is_allowed(method=None, resource=None, *args, **kwargs) -> bool:
...
-    if role is None:
+    if g.user is None:
         name = "unknown"
     else:
-        name = role["name"]
+        role = g.user.role
+        if role is None:
+            name = "unknown"
+        else:
+            name = role["name"]
```
Comment (amihaiemil) [inline timeless/access_control/authorization.py]: @paulodamaso can we simplify this block? I think it's equivalent to 

```python
if g.user is None or g.user.role is None:
    name = "unknown"
else:
    name = role["name"]
```

File: tests/integration/it_employees_test.py [hunk 4]
```diff
@@ -63,9 +69,10 @@ def create_employee():
                     password="bla",
                     pin_code=1234,
                     comment="No comments",
-    )
+                    )
 
 
+@pytest.mark.skip(reason="Authentication injection not implemented")
```
Comment (amihaiemil) [inline tests/integration/it_employees_test.py]: @paulodamaso can you also implement a test with authentication, not only Forbidden cases? So you know it actually works and there will also be an example for later. 

If you don't have anymore time, then maybe add more details to the puzzle above? How should the tests be modified to support authentication?

Comment (paulodamaso) [reply tests/integration/it_employees_test.py]: @amihaiemil I've already left the puzzle for this task, now it's correct :)

**Comment by paulodamaso** [comment]
@amihaiemil Thanks for the review, corrections done, please take a look

**Comment by 0crat** [comment]
@amihaiemil[/z](https://www.0crat.com/u/amihaiemil) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by amihaiemil** [comment]
@rultor good to merge

**Comment by emilianodellacasa** [comment]
@rultor merge

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @amihaiemil[/z](https://www.0crat.com/u/amihaiemil), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#436](https://github.com/timelesslounge/timelessis/issues/436) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>The code reviewer found at least three problems in the code.
Comments were mostly about design problems, not cosmetic issues.

Only one major issue was found during code review. 
Please confirm that you'll try to find at least three major problems while future reviews.

**Comment by ypshenychka** [comment]
@amihaiemil Please see my remark above

**Comment by amihaiemil** [comment]
@ypshenychka I confirm :)

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)

**Comment by 0crat** [comment]
Quality review completed: +6 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request merged**